pkg/master/master_test.go

```diff
@@ -97,6 +97,10 @@ func newMaster(t *testing.T) (*Master, *etcdtesting.EtcdTestServer, Config, *ass
 	config.ProxyDialer = func(network, addr string) (net.Conn, error) { return nil, nil }
 	config.ProxyTLSClientConfig = &tls.Config{}
 
+	// TODO: this is kind of hacky, make testing flakes go away even with
+	//   the sync loop running.
+	config.disableThirdPartyControllerForTesting = true
```

Comment:
I don't understand how this can work, this sets it in the config but the controller checks it in the master?